### PR TITLE
feat: add grace_period and deregistration_delay to http for lb web services.

### LINKED ISSUE #2190: Support configuring `http.deregistration_delay` to reduce connection drain period and speed up deployments
Hey,

When releasing a new version of my service using `svc deploy` I see that the whole process takes around 5 minutes, I suspect that due to the default deregistration_delay value, is there a friendly way to set it in the manifest so it will be set to a lower value?

### LINKED ISSUE #2506: Support HealthCheckGracePeriodSeconds for Load Balanced Web Services
In my manifest I have defined `image.healthcheck.start_period` with a value of 120. But when I look in my ECS service in the details tab it shows this

image

Is this the same setting, or maybe it's just not working?

### PR DESCRIPTION
**Pull request by bvtujo**
Implements two new fields in the manifest under `http`: `grace_period` and `deregistration_delay`. These allow customization of the corresponding fields in the target group health check and service parameters. 

Resolves #2190, resolves #2506 

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: site/content/docs/include/http-config.en.md [hunk 1]
```diff
@@ -22,6 +22,7 @@ http:
     unhealthy_threshold: 2
     interval: 15s
     timeout: 10s
+    deregistration_delay: 60s
```
Comment (iamhopaul123) [inline site/content/docs/include/http-config.en.md]: should we add `grace_period` as well?

**Review by iamhopaul123 — APPROVED** [review]
🚢

File: site/content/docs/include/http-config.en.md [hunk 2]
```diff
@@ -42,6 +43,12 @@ The approximate amount of time, in seconds, between health checks of an individu
 <span class="parent-field">http.healthcheck.</span><a id="http-healthcheck-timeout" href="#http-healthcheck-timeout" class="field">`timeout`</a> <span class="type">Duration</span>  
 The amount of time, in seconds, during which no response from a target means a failed health check. The default is 5s. Range 5s-300s.
 
+<span class="parent-field">http.healthcheck.</span><a id="http-healthcheck-deregistration-delay" href="#http-healthcheck-deregistration-delay" class="field">`deregistration_delay`</a> <span class="type">Duration</span>  
+The amount of time to wait for targets to drain connections during deregistration. The default is 60s. Setting this to a larger value gives targets more time to gracefully drain connections, but increases the time required for new deployments.
```
Comment (Lou1415926) [inline site/content/docs/include/http-config.en.md]: Should we mention that the range is `0-3600` seconds?

Comment (Lou1415926) [reply site/content/docs/include/http-config.en.md]: Also, these are super helpful explanations on the two new fields! Awesome

File: go.sum [hunk 3]
```diff
@@ -1325,6 +1325,7 @@ golang.org/x/tools v0.0.0-20210105154028-b0ab187a4818/go.mod h1:emZCQorbCU4vsT4f
 golang.org/x/tools v0.0.0-20210106214847-113979e3529a/go.mod h1:emZCQorbCU4vsT4fOWvOPXz4eW1wZW4PmDk9uLelYpA=
 golang.org/x/tools v0.1.0/go.mod h1:xkSsbof2nBLbhDlRMhhhyNLN/zl3eTqcnHD5viDpcZ0=
 golang.org/x/tools v0.1.1/go.mod h1:o0xws9oXOQQZyjljx8fwUC0k7L1pTE6eaCbjGeHmOkk=
+golang.org/x/tools v0.1.2 h1:kRBLX7v7Af8W7Gdbbc908OJcdgtK8bOz9Uaj8/F1ACA=
```
Comment (Lou1415926) [inline go.sum]: You probably have run the `go mod tidy` but I just wanted to double check 👀

File: internal/pkg/deploy/cloudformation/stack/testdata/workloads/svc-staging.stack.yml [hunk 4]
```diff
@@ -317,7 +317,7 @@ Resources:
       Protocol: HTTP
       TargetGroupAttributes:
         - Key: deregistration_delay.timeout_seconds
-          Value: 60                  # Default is 300.
+          Value: 30                  # Default is 300.
```
Comment (Lou1415926) [inline internal/pkg/deploy/cloudformation/stack/testdata/workloads/svc-staging.stack.yml]:
```suggestion
          Value: 30 
```

**Review by Lou1415926 — APPROVED** [review]
This is great! Feel free to remove the label once the comments are addressed ⬇️

File: go.sum (hunk 3 above)
Comment (bvtujo) [reply go.sum]: I have!

File: site/content/docs/include/http-config.en.md (hunk 2 above)
Comment (bvtujo) [reply site/content/docs/include/http-config.en.md]: Great call. Fixed.

**Review by iamhopaul123 — APPROVED** [review]

**Review by Lou1415926 — APPROVED** [review]
:shipit:

**Comment by yudori** [comment]
Hey, I noticed the documentation changes made here aren't live yet https://aws.github.io/copilot-cli/docs/include/http-config/ . Maybe an oversight or Is there something I'm missing? @bvtujo @efekarakus

**Comment by efekarakus** [comment]
Hi @yudori ! The feature should be out hopefully this afternoon in v1.9.0 🤞 it's not yet available :)

**Comment by efekarakus** [comment]
@yudori The feature is now released in v1.9.0: https://github.com/aws/copilot-cli/releases/tag/v1.9.0! 🎊

**Comment by yudori** [comment]
@efekarakus You rock!

**Pull request merged**